ui/jobs/bar.d.ts

```diff
@@ -0,0 +1,51 @@
+import { Job } from '../../types/job';
+import { JobDetail } from '../../types/event';
+import ComboTracker, { ComboCallback } from './combo_tracker';
+import TimerBox from '../../resources/timerbox';
+import { kLevelMod } from './constants';
+
+
+export interface Box extends HTMLElement {
+  elapsed: number;
+  duration: number;
+  threshold: number;
+  valuescale: number;
+  fg: string;
+  // parentNode: { classList: ()};
+}
+
+type Selector = {
+  id?: string;
+  fgColor?: string;
+  classList?: string[];
+  threshold?: number;
+};
+
+interface ResourceBox extends HTMLDivElement {
+  parentNode: HTMLElement;
+}
+
+export interface Bars {
+  job: Job;
+  speedBuffs: {
+    circleOfPower: boolean;
+    presenceOfMind: boolean;
+    paeonStacks: number;
+    lightningStacks: number;
+    huton: boolean;
+    shifu: boolean;
+    museStacks: 1 | 2 | 3 | 4;
+  };
+  level: Exclude<Partial<typeof kLevelMod>['length'], (typeof kLevelMod)['length']>;
```

Comment:
This is an amazing construction, but `level` is going to be a `number` when `jobs.js` gets converted to TypeScript, so I think it makes sense to start with that now (sorry).  The level comes from [here](https://github.com/quisquous/cactbot/blob/main/types/event.d.ts#L295).